src/sentry/ingest/billing_metrics_consumer.py

```diff
@@ -222,9 +229,13 @@ def _get_bucket_payload(self, message: Message[KafkaPayload]) -> MetricsBucket:
         payload = json.loads(message.payload.value.decode("utf-8"), use_rapid_json=True)
         return cast(MetricsBucket, payload)
 
-    def _count_processed_transactions(self, bucket_payload: MetricsBucket) -> int:
-        if bucket_payload["metric_id"] != self.metric_id:
+    def _count_processed_transactions(self, org_id: int, bucket_payload: MetricsBucket) -> int:
+        counter_metric_id = indexer.resolve(
+            UseCaseKey.PERFORMANCE, org_id, self.counter_metric_name
+        )
```

Comment:
Initially, I was going to do that. `resolve` and other methods in the indexer require the org id (I assume it's because the mapping can be different from org to org, even if this metric is a common one), so IIUC the resolution must happen for each bucket.

Reply:
Sorry, I forgot about the org id specific resolution. You are right, there is no concept of shared id between organization in the indexer. Tags are per org. The fact that we share a list of them is an implementation detail that, as of today, the product cannot depend on. So resolving for every org is the right option.

Now that has a problem, while you can afford this because that string is hardcoded, this as well is an implementation details of the indexer you should not be depending on. Should that change and should we start storing shared ids in postgres, this consumer would break as it cannot afford to a memcache/postgres roundtrip for each message received. The throughput would plummet. 

I think your idea, of extracting the mapping from the message and comparing the string is probably a better option.